### PR TITLE
Fix a bunch of stuff epubcheck really hated

### PR DESCRIPTION
**Pull request by belldandu**
Uhhhhh i kinda worked from the wrong branch but these are the fixes.

### CODE REVIEW

File: plugin/js/parsers/BakaTsukiParser.js [hunk 1]
```diff
@@ -106,6 +106,14 @@ BakaTsukiParser.prototype.setCoverImage = function (imageInfo) {
 BakaTsukiParser.prototype.removeUnwantedElementsFromContentElement = function (element) {
     let that = this;
     util.removeElements(that.getElements(element, "script"));
+    // Strip headline id of illegal characters that epubcheck doesn't like
+    let headlines = util.getElements(element, "span", e => (e.className === "mw-headline"));
+    for(let hl of headlines){
+        hl.setAttribute("id", util.safeForId(hl.getAttribute("id")));
+    }
+
+    // discord <br /> tags as epubcheck says they are invalid and not allowed in xhtml
```
Comment (belldandu) [inline plugin/js/parsers/BakaTsukiParser.js]: discard* wow that typo

File: plugin/js/EpubPacker.js [hunk 2]
```diff
@@ -219,7 +219,7 @@ EpubPacker.prototype = {
     buildNavPoint: function (parent, playOrder, chapterInfo) {
         let that = this;
         let navPoint = that.createAndAppendChild(parent, "navPoint");
-        navPoint.setAttribute("id", util.zeroPad(playOrder));
+        navPoint.setAttribute("id", that.makeId(util.zeroPad(playOrder)));
```
Comment (belldandu) [inline plugin/js/EpubPacker.js]: Fixes #17

File: plugin/js/parsers/BakaTsukiParser.js [hunk 3]
```diff
@@ -157,11 +165,11 @@ BakaTsukiParser.prototype.stripGalleryBox = function (element) {
         for(let d of util.getElements(listItem, "div")) {
             that.stripWidthStyle(d);
         }
-        that.insertAfter(listItem.parentNode.previousSibling, listItem.firstChild);
-        util.removeNode(listItem);
+        that.insertAfter(util.getElement(element, "h3"), listItem.firstChild);
```
Comment (belldandu) [inline plugin/js/parsers/BakaTsukiParser.js]: Fixes #18

**Comment by belldandu** [comment]
The rest of the changes are miscellaneous changes that reduce errors and warnings in epubcheck

**Comment by belldandu** [comment]
@dteviot Oh yeah i finally got rid of that rougue ul as well as appearantly invalid br tags. Appearantly xhtml does not allow br tags so i removed them all regardless of where they where.
I will actually make a quick push to insert p tags before removing the br's

**Comment by dteviot** [comment]
While you're about it, you may want to add a call to 
util.safeForFileName()  to BakaTsukiImageInfo.prototype.makeZipHref.
Also, now note three cases of the pattern

let chapter = util.safeForFileName(that.chapterTitle);
 +    return "OEBPS/Text/" + util.zeroPad(this.index) + (chapter ? "_" : "") + chapter + ".xhtml";

Maybe have a util.makeStorageFileName(subdirectory, index, title, extension) function?

**Comment by belldandu** [comment]
Yeah and scratch the adding of p tags. epubcheck appearantly just doesn't like where the br's were placed. And it doesn't look like the text was effected by their removal.

Will add that function though @dteviot

**Comment by belldandu** [comment]
@dteviot added. 
Also this is the last remaining issue i can't seem to fix. Probably has something to do with some nav labels being inside navlabels.

```
[kami@Index ~]$ epubcheck Toaru_Maju...dexVolume1.epub 
Validating using EPUB version 2.0.1 rules.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,884): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,1080): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,2006): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,2191): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,2735): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.
ERROR(RSC-005): Toaru_Maju...dexVolume1.epub/OEBPS/toc.ncx(1,2923): Error while parsing file 'different playOrder values for navPoint/navTarget/pageTarget that refer to same target'.

Check finished with errors

epubcheck completed
```

**Comment by belldandu** [comment]
Made an issue for this btw https://github.com/dteviot/WebToEpub/issues/20

**Comment by belldandu** [comment]
@dteviot there no warnings or errors.

**Comment by belldandu** [comment]
@dteviot ready when you are.

**Comment by belldandu** [comment]
Closing as i just now noticed it was remote merged.

**Pull request closed without merging**